### PR TITLE
Handle DebugScope in compact-ids pass

### PR DESCRIPTION
**Pull request by jaebaek**
We have to update OpenCL.DebugInfo.100 DebugScope for instructions if we
change its lexical scope or DebugInlinedAt.

### CODE REVIEW

File: source/opt/compact_ids_pass.cpp [hunk 1]
```diff
@@ -56,6 +56,17 @@ Pass::Status CompactIdsPass::Process() {
           }
           ++operand;
         }
+        if (inst->GetDebugScope().GetLexicalScope() != kNoDebugScope) {
+          auto it =
+              result_id_mapping.find(inst->GetDebugScope().GetLexicalScope());
+          if (it != result_id_mapping.end())
```
Comment (s-perron) [inline source/opt/compact_ids_pass.cpp:62]: In what situations will you not find the lexical scope in the map?

Comment (s-perron) [reply source/opt/compact_ids_pass.cpp:62]: I would suggestion you extract the code in the loop that from lines 37-45 into a new function "GetRemappedId" or something like that.  Then you should call that function here as well.

File: source/opt/compact_ids_pass.cpp [hunk 2]
```diff
@@ -56,6 +56,17 @@ Pass::Status CompactIdsPass::Process() {
...
+            inst->UpdateLexicalScope(it->second);
```
Comment (s-perron) [inline source/opt/compact_ids_pass.cpp:63]: You should be setting modified to true when you change something.  You should only update the value if it is actually a different value.

File: source/opt/compact_ids_pass.cpp [hunk 3]
```diff
@@ -56,6 +56,17 @@ Pass::Status CompactIdsPass::Process() {
...
+        }
+        if (inst->GetDebugInlinedAt() != kNoInlinedAt) {
+          auto it = result_id_mapping.find(inst->GetDebugInlinedAt());
```
Comment (s-perron) [inline source/opt/compact_ids_pass.cpp:66]: All of the same comments as above.

File: source/opt/compact_ids_pass.cpp (hunk 1 above)
Comment (jaebaek) [reply source/opt/compact_ids_pass.cpp]: done

File: source/opt/compact_ids_pass.cpp (hunk 2 above)
Comment (jaebaek) [reply source/opt/compact_ids_pass.cpp]: done

File: source/opt/compact_ids_pass.cpp (hunk 3 above)
Comment (jaebaek) [reply source/opt/compact_ids_pass.cpp]: done

File: source/opt/compact_ids_pass.cpp [hunk 4]
```diff
@@ -21,6 +21,22 @@
 
 namespace spvtools {
 namespace opt {
+namespace {
+
+uint32_t GetRemappedId(
```
Comment (s-perron) [inline source/opt/compact_ids_pass.cpp:26]: Please add comments describing this function.

Comment (jaebaek) [reply source/opt/compact_ids_pass.cpp:28]: Ah right. Done.

**Pull request merged**